### PR TITLE
remove .id restriction for list_to_dic

### LINKED ISSUE #13: /tool romon set id conflict with api .id
##### SUMMARY
RouterOS api use '.id' for configuration item identify, however routeros configuration also contain 'id' as configuration parameter.
For example ` /tool romon set id=`

Currently RouterOS API module have a specific check if 'id' is exactly '.id' in order to be sure will do the correct api call however this do a conflict with routeros configuration parameters.

    "msg": [
        "**'id' must be '.id**'"
    ]

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
community.routeros.api

##### ANSIBLE VERSION
```
ansible 2.10.3
  config file = /opt/gitea/ansible/ansible.cfg
  configured module search path = ['/opt/gitea/ansible/library']
  ansible python module location = /usr/local/lib/python3.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.3 (default, Jul 25 2020, 13:03:44) [GCC 8.3.0]
```

##### STEPS TO REPRODUCE
```
---
- name: configure tool romon
  community.routeros.api:
    hostname: "{{ ros_hostname }}"
    ssl: "{{ ros_ssl }}"
    password: "{{ ros_password }}"
    username: "{{ ros_username }}"
    path: "{{ ros_tool_romon_path }}"
    cmd:
      set
      enabled={{ ros_tool_romon_enabled }}
      secrets={{ ros_tool_romon_secrets }}
      id={{ ros_tool_romon_id }}
  register: toolromon
  delegate_to: localhost

- name: result configure tool romon
  debug:
    msg: "{{ toolromon }}"
```

##### EXPECTED RESULTS
no msg should be return 
`"msg": []`

##### ACTUAL RESULTS
```ok: [ros1] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "add": null,
            "cmd": "set enabled=yes secrets= id=00:00:00:00:00:00",
            "hostname": "ros1",
            "password": "",
            "path": "tool romon",
            "port": null,
            "query": null,
            "remove": null,
            "ssl": true,
            "update": null,
            "username": "admin"
        }
    },
    "msg": [
        "'id' must be '.id'"
    ]
}
```

### PR DESCRIPTION
**Pull request by NikolayDachev**
##### SUMMARY
Remove .id restriction, we have separate checks already (where is needed)
Fixes #13

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
community.routeros.api

### CODE REVIEW

**Comment by NikolayDachev** [comment]
Actually I'm not sure for change log and if we should bump the version

**Comment by felixfontein** [comment]
@NikolayDachev don't wory about the version, I'll handle that during the release process. I can make a 1.0.1 patch release after this is merged, since we don't have anything waiting that would require a minor release.

About changelog fragment: simply create one describing this bugfix, the general instructions on how they should look like are here: https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to

**Review by heuels — APPROVED** [review]
Looks good, thanks!

File: changelogs/fragments/13-remove-id-restriction-for-api.yaml
Comment (felixfontein) [inline changelogs/fragments/13-remove-id-restriction-for-api.yaml:2]:
```suggestion
- api - remove ``id to .id`` as default requirement which conflict with routeros ``id`` configuration parameter (https://github.com/ansible-collections/community.routeros/pull/15).
```

Comment (NikolayDachev) [reply changelogs/fragments/13-remove-id-restriction-for-api.yaml]: Thanks!!!

**Comment by felixfontein** [comment]
@heuels @NikolayDachev I'll create a new release for the collection somewhen tomorrow or over the weekend.

Feel free to merge this once tests pass, otherwise I'll do it tomorrow. I'm off to bed now ;)

**Pull request merged**
